### PR TITLE
Add test using tap framework

### PR DESCRIPTION
**Pull request by nechaido**
Due to request from @aqrln in #79 I open a PR for reviewing tests and choosing test framework.

### CODE REVIEW

File: test/array.find.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+tap.test('find', (test) => {
+  const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+  const expected = [15];
+  const predicate = (item, callback) => process.nextTick(() => {
+    callback(null, item % 3 === 0 && item % 5 === 0);
+  });
+
+  metasync.find(data, predicate, (err, result) => {
+    test.error(err, 'must not return an error');
+    test.ok(result, 'result must be initialized');
```
Comment (aqrln) [inline test/array.find.js]: The next check covers it.

File: test/array.find.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    test.type(result, 'Array', 'type of result must be array');
+    test.strictSame(result.length, expected.length,
+      `length of result must be ${expected.length}`);
+    test.strictSame(result[0], expected[0], `reult should be: ${expected}`);
```
Comment (aqrln) [inline test/array.find.js]: Can you please replace these four lines with a single `test.strictSame(result, expected, `...`)`? Also there's a typo in the test message ("re**s**ult").

File: test/chain.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  const fn = (item, callback) => process.nextTick(() => {
+    callback(null, item * 2);
+  });
+
+  metasync.for(data).map(fn).fetch((err, result) => {
+    test.error(err, 'must not return an error');
+    test.type(result, 'Array', 'type of result must be array');
+    test.strictSame(result.length, expected.length,
+      `length of result must be ${expected.length}`);
+    test.strictSame(result, expected, `reult should be: ${expected}`);
```
Comment (aqrln) [inline test/chain.js]: There's no need to check the type and length, this test covers them.

File: package.json [hunk 4]
```diff
@@ -21,14 +21,15 @@
...
+    "test": "tap -b -t 1000 test/*.js && npm run lint",
     "lint": "eslint ."
   },
   "engines": {
     "node": ">=6.0.0"
   },
   "dependencies": {
-    "metarhia-common": "0.0.x"
+    "metarhia-common": "0.0.x",
+    "tap": "^10.0.0"
```
Comment (aqrln) [inline package.json]: This must be a `devDependency`, not a `dependency`. Forcing the users of the library to install `tap` is atrociously cruel given its package size, not to mention completely unnecessary.

File: package.json [hunk 5]
```diff
@@ -21,14 +21,15 @@
   "main": "./metasync.js",
   "readmeFilename": "README.md",
   "scripts": {
-    "test": "node tests/examples && node tests/chain && npm run lint",
+    "test": "tap -b -t 1000 test/*.js && npm run lint",
```
Comment (aqrln) [inline package.json]: See https://github.com/metarhia/MetaSync/pull/99#discussion_r113441182

**Comment by tshemsedinov** [comment]
@aqrln & @DzyubSpirit let's land tap ?

**Comment by DzyubSpirit** [comment]
@nechaido Is it okay that tests fail? Is that problem because of functionality or tests?

**Comment by tshemsedinov** [comment]
@DzyubSpirit you can see it in Travis here: https://travis-ci.org/metarhia/MetaSync/builds/226031887
For example: https://travis-ci.org/metarhia/MetaSync/jobs/226031888

**Comment by nechaido** [comment]
@DzyubSpirit it fails because `find()` function was broken at the moment of writing this test, it won't after merging new commits from master

**Comment by aqrln** [comment]
@nechaido let's rebase and see, then :)

**Comment by nechaido** [comment]
@aqrln already

File: test/array.find.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+tap.test('find', (test) => {
+  const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16];
+  const expected = 15;
+  const predicate = (item, callback) => process.nextTick(() => {
+    callback(null, item % 3 === 0 && item % 5 === 0);
+  });
+
+  metasync.find(data, predicate, (err, result) => {
+    test.error(err, 'must not return an error');
+    test.strictSame(result, expected, `reult should be: ${expected}`);
```
Comment (aqrln) [inline test/array.find.js]: typo: re**s**ult

File: test/chain.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+tap.test('map', (test) => {
+  const data = [1, 2, 3, 4];
+  const expected = [2, 4, 6, 8];
+  const fn = (item, callback) => process.nextTick(() => {
+    callback(null, item * 2);
+  });
+
+  metasync.for(data).map(fn).fetch((err, result) => {
+    test.error(err, 'must not return an error');
+    test.strictSame(result, expected, `reult should be: ${expected}`);
```
Comment (aqrln) [inline test/chain.js]: ditto

**Comment by nechaido** [comment]
@tshemsedinov ping

**Comment by tshemsedinov** [comment]
Landed in 72ed1dcbf903d0307419167e54f87d6697bba384, ec66cd6fd37f152bce8d03226dcd56d06ac150e5 and 962e76ada1c09d66271490caef46c5208ee91811

**Pull request closed without merging**